masonry/src/core/contexts.rs

```diff
@@ -257,6 +257,41 @@ impl MutateCtx<'_> {
         }
     }
 
+    /// Return a [`WidgetMut`] to a child widget by its id.
+    pub fn find_mut(&mut self, child_id: WidgetId) -> WidgetMut<'_, dyn Widget> {
+        let child_state_mut = self
+            .widget_state_children
+            .reborrow_mut()
+            .find_mut(child_id)
+            .expect("find_mut: child not found");
```

Comment:
So, correct me if I'm missing something, but the problem with that implementation is that WidgetMut will not correctly update the flags of any intermediary nodes between the current widget and the descendant widget.

Like, say you have a `A -> B -> C -> D -> E` chain and you do `A.ctx.find_mut(E)` and then update some stuff; the flags for A and E will be updated, but not the flags for B, C and D.

You need something like #1011 to make this method possible.

Reply:
Oh, right yeah. Forgot about that.